### PR TITLE
[Fix] fix max stake for single

### PR DESCRIPTION
**Pull request by camfairchild**
This PR addresses a bug with max stake that effected staking with only a single hotkey `--wallet.hotkey hk0` using the `--max_stake <#>` flag.  

e.g. `btcli stake --wallet.hotkey hk0 --max_stake 1.0`

### CODE REVIEW

File: bittensor/_cli/cli_impl.py [hunk 1]
```diff
@@ -347,9 +347,7 @@ def stake( self ):
                 bittensor.wallet( config = self.config, hotkey = hotkey ) for hotkey in self.config.wallet.get('hotkeys')
             ]
         else:
-            # Do regular stake
-            subtensor.add_stake( wallet, amount = None if self.config.get('stake_all') else self.config.get('amount'), wait_for_inclusion = True, prompt = not self.config.no_prompt )
-            return None
+            wallets_to_stake_to = [ bittensor.wallet( config = self.config ) ]
```
Comment (Eugene-hu) [inline bittensor/_cli/cli_impl.py:353]: I am confused about the previous implementation. So the `else` only covers the case in which there is a single wallet to stake too?  If so, we should be more specific.

File: bittensor/_cli/cli_impl.py [hunk 2]
```diff
@@ -384,6 +383,11 @@ def stake( self ):
             final_amounts.append(stake_amount_tao)
             final_wallets.append(wallet)
 
+        if len(final_wallets) == 0:
+            # No wallets to stake to.
+            bittensor.__console__.print("Not enough balance to stake to any hotkeys or max_stake is less than current stake.")
```
Comment (Eugene-hu) [inline bittensor/_cli/cli_impl.py:391]: The console error should be updated I think

File: bittensor/_cli/cli_impl.py [hunk 3]
```diff
@@ -374,7 +372,8 @@ def stake( self ):
             stake_amount_tao: float = self.config.get('amount')
             if self.config.get('max_stake'):
                 wallet_stake: Balance = wallet.get_stake()
-                stake_amount_tao: float =  self.config.get('max_stake') - wallet_stake.tao
+                stake_amount_tao: float = self.config.get('max_stake') - wallet_stake.tao
```
Comment (Eugene-hu) [inline bittensor/_cli/cli_impl.py:378]: I see, so previously, `max_stake` never worked if there was only a single wallet

**Review by Eugene-hu — APPROVED** [review]
Nice job!

File: bittensor/_cli/cli_impl.py (hunk 2 above)
Comment (camfairchild) [reply bittensor/_cli/cli_impl.py:391]: Hotkeys (wallets) are removed when we don't have enough balance to stake the amount to them.

**Pull request merged**